### PR TITLE
Add + apply grade grooming helpers

### PR DESCRIPTION
**Pull request by sharanya204**

### CODE REVIEW

**Comment by sharanya204** [comment]
before
after

~TODO (right now): make this sport only~ done

image

**Comment by lukehtravis** [comment]
Amaaaazing!!!

**Review by lukehtravis — APPROVED** [review]

**Comment by sharanya204** [comment]
image
roped and boulders are separated!

File: app/dashboard/components/Form.tsx [hunk 1]
```diff
@@ -17,34 +18,48 @@ const Form: React.FC<Props> = ({populateData}: Props) => {
...
+            const ropedOnlyNaive = resultsData.filter((oneRoute) => roped.includes(oneRoute["Route Type"]));
+            ropedOnlyNaive.forEach(element => {
+              element.Rating = eliminateSlashesFromGrades(element.Rating);
+              element.Rating = flattenPlusAndMinusGrades(element.Rating);
+              element.Rating = removeRiskRating(element.Rating);
+              console.log(element.Rating)
+              // soundas: this seems a bit kludgy; TODO: how do I chain + assign concisely
+            });
+            const boulderOnlyNaive = resultsData.filter((oneRoute) => boulder.includes(oneRoute["Route Type"]) || boulderStyleNaive.includes(oneRoute["Style"]));
+            populateData(ropedOnlyNaive) // how...will populate now work. Luke, help!
```
Comment (sharanya204) [inline app/dashboard/components/Form.tsx]: I guess...I'm just going to pass ropedOnlyNaive to its own viz functions and boulderOnlyNaive to its own viz functions? 

@lukehtravis I need some structural oversight. help.

Comment (lukehtravis) [reply app/dashboard/components/Form.tsx]: You can recombine the arrays by doing

populateData([…ropedOnlyNaive, …boulderOnlyNaive])

The ellipse here is called the spread operator in javascript and it concisely joins arrays…we push all data into populateData, then can filter for bouldering data later on the vis side of things

https://www.w3schools.com/react/react_es6_spread.asp

**Pull request merged**